ethereum/eth/src/main/java/tech/pegasys/pantheon/ethereum/eth/sync/worldstate/WorldStateDownloader.java

```diff
@@ -146,6 +146,13 @@ private void requestNodeData(final BlockHeader header) {
             if (pendingRequest == null) {
               break;
             }
+            final Optional<BytesValue> existingData =
+                pendingRequest.getExistingData(worldStateStorage);
+            if (existingData.isPresent()) {
+              pendingRequest.setData(existingData.get());
+              queueChildRequests(pendingRequest);
+              continue;
```

Comment:
Looks like you may need to change the containing `for` loop to a `while` loop so that we don't send out smaller than intended requests when we run into known nodes.